practice/tasks/IDisposable/IDisposableImplementation/IDisposableImplementation/MemoryStreamLogger.cs

```diff
@@ -1,22 +1,51 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
 
 namespace NetMentoring
 {
-    public class MemoryStreamLogger
+    public class MemoryStreamLogger : IDisposable
     {
-        private FileStream memoryStream;
-        private StreamWriter streamWriter;
+        private static FileStream _memoryStream;
+        private static StreamWriter _streamWriter;
+        private const string PathToFile = @"Files\\log.txt";
+        private bool _disposed;
 
-        public MemoryStreamLogger()
+
+        public void ReadFile()
         {
-            memoryStream = new FileStream(@"\log.txt", FileMode.OpenOrCreate);
-            streamWriter = new StreamWriter(memoryStream);
+            using (var file = new StreamReader(PathToFile))
+            {
+                var counter = 0;
+                string line;
+                while ((line = file.ReadLine()) != null)
+                {
+                    Console.WriteLine(line);
+                    counter++;
+                }
+                file.Close();
+                Console.WriteLine($"File has {counter} lines.");  
+            }
         }
 
         public void Log(string message)
         {
-            streamWriter.Write(message);
+            _memoryStream = new FileStream(PathToFile, FileMode.Open);
+            _streamWriter = new StreamWriter(_memoryStream);
+            _streamWriter.Write(message);
+        }
+
+        public void Dispose()
+        { 
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
what happens if we skip this row? did you try to work with Destructor/IFinalize?

Reply:
we commented it and check yesterday, we go to the destructor in this case